crates/nil/src/vfs.rs

```diff
@@ -245,13 +251,17 @@ impl LineMap {
     }
 
     pub fn end_col_for_line(&self, line: u32) -> u32 {
-        let mut len = self.line_starts[line as usize + 1] - self.line_starts[line as usize];
+        let mut len = if line + 1 >= self.line_starts.len() as u32 {
+            self.len - self.line_starts[line as usize]
+        } else {
+            self.line_starts[line as usize + 1] - self.line_starts[line as usize]
```

Comment:
The condition below can be merged into here.
```suggestion
            // Minus the trailing `\n` for non-last-lines.
            self.line_starts[line as usize + 1] - self.line_starts[line as usize] - 1
```